src/spikeinterface/generation/template_database.py

```diff
@@ -1,12 +1,140 @@
-from spikeinterface.core.template import Templates
 import zarr
+import functools
+import numpy as np
+
+from spikeinterface.core.template import Templates
+
 
+@functools.cache
+def fetch_template_object_from_database(dataset="test_templates.zarr") -> Templates:
+    """
+    Fetch a template dataset from the spikeinterface template database.
+    A dataset is a collection of templates with associated metadata for one specific recording.
 
-def fetch_templates_from_database(dataset="test_templates.zarr") -> Templates:
+    Parameters
+    ----------
+    dataset : str, default: "test_templates"
+        The name of the dataset to fetch.
+        The dataset must be available in the spikeinterface template database.
 
+    Returns
+    -------
+    Templates
+        _description_
+    """
     s3_path = f"s3://spikeinterface-template-database/{dataset}/"
     zarr_group = zarr.open_consolidated(s3_path, storage_options={"anon": True})
 
     templates_object = Templates.from_zarr_group(zarr_group)
 
     return templates_object
+
+
+@functools.cache
+def fetch_templates_database_info() -> "pandas.DataFrame":
+    """
+    Fetch the information about the templates in the spikeinterface template database.
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe containing the template information.
+    """
+    import pandas as pd
+
+    s3_path = "s3://spikeinterface-template-database/templates.csv"
+    df = pd.read_csv(s3_path, storage_options={"anon": True})
+
+    return df
+
+
+def list_available_datasets_in_template_database() -> list:
+    """
+    List all available datasets in the spikeinterface template database.
+
+    Returns
+    -------
+    list
+        List of available datasets.
+    """
+    df = fetch_templates_database_info()
+    datasets = np.unique(df["dataset"]).tolist()
+
+    return datasets
+
+
+def query_templates_from_database(
+    template_df_or_indices: "pandas.DataFrame | list[int] | np.ndarray", verbose: bool = False
+) -> Templates:
+    """
+    Retrieve templates from the spikeinterface template database.
+
+    Parameters
+    ----------
+    template_df_or_indices : pd.DataFrame or array-like
+        Dataframe containing the template information, obtained by slicing/querying the output of fetch_templates_info.
+
+    Returns
+    -------
+    Templates
+        The templates object.
+    """
+    import pandas as pd
+
+    templates_array = []
+    if isinstance(template_df_or_indices, pd.DataFrame):
+        template_info = template_df_or_indices
+    else:
+        template_info = fetch_templates_database_info().iloc[template_df_or_indices]
```

Comment:
I think we should removed this option as it changes when reconstructing the database. Or have a global id as @samuelgarcia suggested. Actually that would probably be the most stable thing.

Reply:
let's remove this option